atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/ConflictDetectionManager.java

```diff
@@ -17,45 +17,57 @@
 
 import java.util.Map;
 
+import com.google.common.base.Supplier;
+import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Maps;
 import com.palantir.atlasdb.keyvalue.api.TableReference;
 import com.palantir.atlasdb.transaction.api.ConflictHandler;
 
 public class ConflictDetectionManager {
-    private final RecomputingSupplier<Map<TableReference, ConflictHandler>> supplier;
+    private final Supplier<Map<TableReference, ConflictHandler>> supplier;
     private final Map<TableReference, ConflictHandler> overrides;
+    private final RecomputingSupplier<Map<TableReference, ConflictHandler>> cachedResult;
 
-    public ConflictDetectionManager(RecomputingSupplier<Map<TableReference, ConflictHandler>> supplier) {
+    /**
+     * @param supplier A {@link Supplier} which returns the current conflict detection
+     *                 configuration to the database. Note that the result is cached,
+     *                 so changes to configuration must be followed by a call to
+     *                 {@link #recompute()} in order to be picked up.
+     */
+    ConflictDetectionManager(Supplier<Map<TableReference, ConflictHandler>> supplier) {
         this.supplier = supplier;
         this.overrides = Maps.newConcurrentMap();
+        this.cachedResult = RecomputingSupplier.create(() -> {
+            ImmutableMap.Builder<TableReference, ConflictHandler> ret = ImmutableMap.builder();
+            ret.putAll(this.supplier.get());
+            ret.putAll(overrides);
+            return ret.build();
```

Comment:
if overrides actually does override a key from the supplied map, `build()` will throw here. I Think we have to putAll both into a HashMap, then either wrap with i modifiable or ImmutableMap.copyOf. We should also add unit tests covering overlapping and disjoint overrides.